test/cfa/ntp_conf_test.rb

```diff
@@ -0,0 +1,305 @@
+require_relative "../test_helper"
+require "cfa/memory_file"
+require "cfa/ntp_conf"
+
+def ntp_disk_file
+  path = File.expand_path("../../fixtures/cfa/ntp.conf", __FILE__)
+  File.read(path)
+end
+
+describe CFA::NtpConf do
+  subject(:ntp) { described_class.new(file_handler: ntp_file) }
+
+  let(:ntp_file) { CFA::MemoryFile.new(content) }
+
+  before do
+    ntp.load
+  end
+
+  describe "#load" do
+    context "when there is only one entry of a collection" do
+      let(:content) do
+        "server 127.127.1.0\n"
+      end
+
+      it "fixes the key of the collection" do
+        expect(ntp.records.first.augeas[:key]).to eq("server[]")
+      end
+    end
+  end
+
+  describe "#records" do
+    let(:content) { ntp_disk_file }
+
+    it "obtains the corrent amount of records" do
+      expect(ntp.records.count).to eq(12)
+    end
+
+    it "obtains a collection of records" do
+      expect(ntp.records).to be_a(CFA::NtpConf::RecordCollection)
+    end
+  end
+end
+
+describe CFA::NtpConf::RecordCollection do
+  subject(:ntp) { CFA::NtpConf.new(file_handler: ntp_file) }
+
+  let(:ntp_file) { CFA::MemoryFile.new(ntp_disk_file) }
+
+  let(:new_record) do
+    record = CFA::NtpConf::ServerRecord.new
+    record.value = "4.pool.ntp.org"
+    record
+  end
+
+  let(:existing_record) { ntp.records.first.dup }
+
+  before do
+    ntp.load
+  end
+
+  context "#add" do
+    context "when does not exist the record to add" do
+      it "adds the record" do
+        ntp.records.add(new_record)
+        expect(ntp.records.include?(new_record)).to be(true)
```

Comment:
to be consistent I think it would be nice if it is there also count to ensure it is there only once.